pkg/controllers/repositoryserver/repositoryserver_controller.go

```diff
@@ -96,6 +97,43 @@ func (r *RepositoryServerReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		r.Recorder.Event(repoServerHandler.RepositoryServer, corev1.EventTypeWarning, "Failed", err.Error())
 		return ctrl.Result{}, err
 	}
+
+	logger.Info("Connect to Kopia Repository")
+	if err := repoServerHandler.connectToKopiaRepository(); err != nil {
+		repoServerHandler.RepositoryServer.Status.Progress = crkanisteriov1alpha1.Failed
+		if uerr := r.Status().Update(ctx, repoServerHandler.RepositoryServer); uerr != nil {
+			return ctrl.Result{}, uerr
+		}
+		return ctrl.Result{}, err
+	}
+
+	logger.Info("Start Repository Server")
+	if err := repoServerHandler.startRepoProxyServer(ctx); err != nil {
+		repoServerHandler.RepositoryServer.Status.Progress = crkanisteriov1alpha1.Failed
+		if uerr := r.Status().Update(ctx, repoServerHandler.RepositoryServer); uerr != nil {
+			return ctrl.Result{}, uerr
+		}
+		return ctrl.Result{}, err
+	}
+
+	logger.Info("Add/Update users")
+	if err := repoServerHandler.createOrUpdateClientUsers(ctx); err != nil {
+		repoServerHandler.RepositoryServer.Status.Progress = crkanisteriov1alpha1.Failed
+		if uerr := r.Status().Update(ctx, repoServerHandler.RepositoryServer); uerr != nil {
+			return ctrl.Result{}, uerr
+		}
+		return ctrl.Result{}, err
+	}
+
+	logger.Info("Refresh Server")
+	if err := repoServerHandler.refreshServer(ctx); err != nil {
+		repoServerHandler.RepositoryServer.Status.Progress = crkanisteriov1alpha1.Failed
+		if uerr := r.Status().Update(ctx, repoServerHandler.RepositoryServer); uerr != nil {
+			return ctrl.Result{}, uerr
+		}
```

Comment:
These lines are getting called repeatedly in this function, can we move it to another function and call it

Reply:
added new function for it.